### PR TITLE
[RDY] Firearm Damage Modifier Rebalance

### LINKED ISSUE #15752: Arbitrary Gun Damage, plus Proposal to Fix
I'm not sure whether this is a remnant from when Whales was developing this (I'm guessing it is), or if it's something more recent, but it's something that's bugged me regarding firearms since I first started playing. And as we slide closer and closer to a more realistic depiction of firearms, it stands out more and more.

I'm referring (as the title states) to the arbitrary way that gun damage values have been set. Right now, each gun seems to have been slapped with a specific damage value, no matter how little it actually made sense; as best I can tell, it seemed to be based on how powerful the person statting it out _thought_ the gun should be, without basing it on any real world data. To the best of my knowledge (and I'll confess here; I'm speaking entirely from looking up statistics online, rather than real world experience), **when it comes to the gun itself**, the only thing that can affect the velocity or energy of the bullet is the length of the barrel; and even then, it's a small difference. A 5.56 NATO fired from an M4 Carbine only loses about 7% of its velocity when compared to that same round fired from an M16, despite being a good quarter shorter.

To give an example of what I'm talking about, let's take a look at two in-game firearms: the M1911 Pistol and the KRISS Vector. The former has a 5.03 inch barrel (for the government model), the latter comes standard with a 5.5 inch barrel. So, all other things being equal, the KRISS Vector should be negligibly more powerful, right? Not according to the game: the M1911 gets an **8 point damage bonus** above the KRISS Vector (the M1911 gets a whopping +6 damage bonus, while the KRISS Vector gets a -2 for some reason).

They aren't the only ones with weird damage values; the TEC-9 (+3, despite being shorter than some 9mm handguns) and Five-seveN (+2, where the P90 gets a flat 0) are two other prominent examples that stick out in my mind, and I'm sure they're not the only ones.

What I'd like to do is go through all the guns and rebalance all of their damage values to something more realistic (or at least internally consistent). What I have in mind currently is to set a baseline firearm for each caliber (9mm = Beretta M9, .45 ACP = M1911, .223 = M16, etc.), which would be considered Damage 0. For each 2 inches in barrel length other firearms differed, it'd provide a plus or minus 1 point difference in damage, rounded to the nearest damage bonus. Revisiting M16 vs. M4, this means the M4 would have a ranged damage of -3 (5.5 inches shorter, rounded to 6 inches), and the M1911 and KRISS Vector would be on equal footing in terms of damage.

This does have the drawback of potentially homogenizing a lot of guns in terms of damage (particularly handguns), but as far as I can tell, the gun used has more of a bearing on accuracy and felt recoil than it does on actual lethal potential; that's decided mostly by the ammo used.

Any thoughts, opinions, or things I've missed? As I stated earlier, my knowledge of firearms is purely theoretical, so there might be something I've overlooked.

### PR DESCRIPTION
**Pull request by Nioca**
Closes #15752

As it currently stands, guns appears to have damage modifiers set purely by numbers out of thin air; this results in weird cases where a bullet fired from a handgun can drastically outperform the same bullet fired from a full-size carbine or SMG (M1911 vs. TDI Vector or Thompson SMG, for example). In real life, to the best of my knowledge, the only aspect of a gun that can significantly effect the damage (in loose terms) a bullet could potentially do is the length and quality of the gun's barrel. In simple terms, all other things being equal, a round fired from a short-barreled firearm is going to have less energy than one fired from a long-barreled firearm, simply because it has less time to accelerate. Barrel quality is pretty much a non-issue, except in the cases of scrap weapons and improvised firearms.

Thus, this rebalance bases the damage of weapons on the length of the weapons' barrels. Each caliber has a baseline set for it, based on the data Stalblar pulled for muzzle energy figures (from Wikipedia, for the curious; there's probably better sources to use, but I'm going to stick with this right now for consistency's sake). For every three inches a firearm deviates from the baseline, it gains a one-point bonus or penalty, rounding to the nearest bonus/penalty (so a gun with 6 Inch barrel with a baseline of 4 Inches would round up to a +1).

In general, the net result of this is that the wide and arbitrary spread on most firearms is narrowed (which helps make firearm selection less of a no-brainer). In addition, larger and bulkier weapons will tend to have better damage, offering an incentive to use an SMG over a pistol, or a machine gun over a compact automatic carbine. The Leadworks line tends to get nerfed a bit, although they're still potent weapons. With scrap weapons, it's a toss-up; the rifle caliber ones tend to get nerfed, but the pistol caliber ones tend to receive damage buffs or stay about where they are.
## Pistol-Caliber Firearms

A few cartridges got a little more variety, with .22 LR getting a nice little buff here, but the main benefit here lies with the 9x19mm and .45 ACP guns. 9mm had its damage basically overhauled, and while the result isn't really visible on the table below, it now makes far more sense gun-to-gun. As for .45 ACP, I'll let the 6 point reduction in spread speak for itself. Overall, pistol-caliber weapons as a whole experienced a 5 point reduction in damage modifier spread.

**DmgMod Range (New/Old)** is the range of damage modifiers on the guns, with a comparison of after and before. **Crt. Range (New/Old)** is that same range with the cartridge's actual damage applied, giving an idea of the ''actual'' damage spread when using guns from that caliber.

| Cartridge | Base Length | DmgMod Range (New/Old) | Crt. Range (New/Old) | Notes |
| --- | --- | --- | --- | --- |
| **OVERALL** |  | **-2 to +5 / -4 to +8** |  |  |
| 4.6x30mm | 7.1 Inches | -1 to 0 / 0 to +2 | 17-18 / 18-20 |  |
| .22 LR | 6 Inches | -2 to +5 / -2 to +2 | 10-17 / 10-14 | The baseline I had to pull from a different source, but it seemed to match up close enough to be servicable. |
| 5.7x28mm | 10.4 Inches | -2 to 0 / 0 to +2 | 18-20 / 20-22 |  |
| .32 ACP | 4 Inches | 0 / 0 | 16 / 16 | No changes. |
| .38 Special | 4 Inches | -2 to 0 / 0 | 20-22 / 22 |  |
| 9x18mm | 3.8 Inches | 0 / 0 | 16 / 16 | No changes. |
| 7.62x25mm | 4.7 Inches | 0 to +2 / 0 to +1 | 15-17 / 15-16 |  |
| 9x19mm | 5.9 Inches | -1 to +4 / -3 to +3 | 25-30 / 23-29 |  |
| .40 S&W | 4 Inches | -2 to +1 / 0 | 26-29 / 28 |  |
| .45 ACP | 5 Inches | 0 to +4 / -4 to +6 | 28-32 / 24-34 |  |
| .44 Magnum | 7.5 Inches | -1 to 0 / 0 | 39-40 / 40 |  |
| .454 Casull | 7.5 Inches | 0 / +6 | 56 / 62 | Cartridge note: it seems Stalblar put the data up, but didn't actually change this round, resulting in a odd discrepency in damage. |
| .500 S&W | 8.4 Inches | 0 / +8 | 48 / 56 | See .454 Casull above. |

And then a breakdown of each gun. The big winners here are .22 LR long-barrel weapons (which got a very nice boost), the Calico M960 (damage penalty axed, got a slight buff), MAC-10 (ditto), and the pistol carbines which simply received a sizable damage buff; the L1820 Long Ranger, Kel-Tec SUB-2000 (which had its name fixed) and Cx4 Storm are now dangerous in their own right, rather than just volume hogs. On the nerf side of things, the TEC-9, Leadworks Autorevolvers, M1911, Taurus Raging Bull, and S&W 500 all had their ridiculous damage bonuses axed almost entirely; most are now 0 damage weapons, with the autorevolvers squeaking into a +1.

| Weapons | Length (in) | NewDmg | OldDmg | Notes |
| --- | --- | --- | --- | --- |
| **4.6x30mm** | **Base: 7.1** |  |  |  |
| H&K UCP | 5.1 Inches | -1 | +2 |  |
| H&K MP7 | 7.1 Inches | 0 | 0 |  |
| **5.7x28mm** | **Base: 10.4** |  |  |  |
| Five-seveN | 4.8 Inches | -2 | +2 |  |
| FN P90 | 10.4 Inches | 0 | 0 |  |
| **.22 LR** | **Base: 6** |  |  |  |
| Ruger LCR .22 | 1.9 Inches | -2 | 0 | Technically still within a -1, but we're talking a ridiculously dinky barrel here; getting **this** small, muzzle energy differences get more drastic. |
| Mossberg Brownie | 2.5 Inches | -1 | 0 |  |
| SIG Mosquito | 3.9 Inches | -1 | 0 |  |
| S&W 22A | 5.5 Inches | 0 | 0 |  |
| Pipe Rifle .22 | 18 Inches? | +1 | -2 | It's a piece of scrap, but it would provide a far longer barrel than most .22 LR pistols. |
| RM360 Rotary Carbine | 15 Inches? | +3 | 0 | It's smaller in volume than the American-180, suggesting a smaller barrel. |
| American 180 | 18.5 Inches | +4 | +2 |  |
| Ruger 10/22 | 18.5 Inches | +4 | 0 |  |
| Marlin 39A | 24 Inches | +5 | 0 | Technically a +6, but you're only going to get so much extra performance out of a long barrel, so let's cap it at +5. |
| **.32 ACP** | **Base: 4** |  |  | No changes here; skipping. |
| **.38 Special** | **Base: 4** |  |  |  |
| Ruger LCR .38 | 1.9 Inches | -2 | 0 | Like the .22 version, technically only a -1, but we're talking a ridiculously dinky barrel here; when you start getting THIS small, the differences in muzzle energy get more drastic |
| 2 Shot Special | Unknown | -1 | 0 | It's hard to say just how big this is, although it's (very large) pistol sized. But it is a handmade weapon. |
| COP .38 | 3 Inches | 0 | 0 |  |
| S&W 619 | 4 Inches | 0 | 0 |  |
| Taurus Pro .38 | 4 Inches | 0 | 0 | Seeing as Taurus, far as I can tell, has never actually manufactured a semiauto pistol in .38, I'm just going to set this at baseline. |
| **9x18mm** | **Base: 3.8** |  |  | No changes here; skipping. |
| **7.62x25mm** | **Base: 4.7** |  |  |  |
| Tokarev TT-30 | 4.6 Inches | 0 | 0 |  |
| PPSh-41 | 10.6 Inches | +2 | +1 |  |
| **9x19mm** | **Base: 5.9** |  |  |  |
| Glock 19 | 4.1 Inches | -1 | 0 |  |
| USP 9mm | 4.3 Inches | -1 | 0 |  |
| Beretta M9 | 4.9 Inches | 0 | -1 |  |
| L39B & L39 | 4.9 Inches | 0 | +1 | The L39s are based off the Auto-9 in Robocop, which is in turn a dressed-up Beretta 93R. |
| TEC-9 | 5 Inches | 0 | +3 | It's assumed here that the in-game TEC-9 is the full-size model and not the mini. |
| Briefcase SMG | Unknown | 0 | +1 | It's comparable in size to an MP5, but its concealed nature likely means a shorter barrel. |
| Lookout 2032 Autorevolver | 7.5 Inches? | +1 | +5 | It's Volume 2, same as the Beretta M9. Autorevolvers tended to vary between 6-7.5 inches in barrel length, so we'll err on the side of a +1 here. |
| STEN | 7.7 Inches | +1 | +1 |  |
| H&K MP5 | 8.9 Inches | +1 | +1 |  |
| Uzi | 10.2 Inches | +1 | 0 |  |
| 9mm Pipe Rifle | Unknown | +1 | -2 | Despite its quality, it IS a small rifle. However, given the far poorer construction, it's unlikely to hit the same velocity or energy as an actual rifle. |
| Calico M960 | 13 Inches | +2 | -3 | Probably also needs its volume increased by 1. It's about the same overall size as an MP5. |
| Kel-Tec SUB-2000 | 16.1 Inches | +3 | 0 |  |
| Cx4 Storm | 16.6 Inches | +4 | 0 |  |
| **40 S&W** | **Base: 4** |  |  |  |
| Survivor's Six Shooter | Unknown | 0 | 0 | It's a handmade scrap weapon, albeit of better design. It IS a bit bigger than the other .40 S&W guns present. |
| SIG Pro .40 | 3.9 Inches | 0 | 0 |  |
| S&W 610 | 6.5 Inches | +1 | 0 |  |
| Glock 22 | 8 Inches | +1 | 0 |  |
| **.45 ACP** | **Base: 5** |  |  |  |
| MAC-10 | 4.49 Inches | 0 | -4 |  |
| M1911 | 5.03 Inches | 0 | +6 |  |
| TDI Vector | 5.5 Inches | 0 | -2 |  |
| L39B-45 & L39-45 | 5.9 Inches? | 0 | +1 | It's Volume 3, bigger than the 1911, but it's based on a scaled up Auto-9 (in turn based on the Beretta 93R), which makes it difficult to be sure. But a 2 inch difference is likely. |
| L2031 Enforcer | 7.5 Inches? | +1 | +5 | It's Volume 2, same as the 1911. Real-life Autorevolvers tended to vary between 6-7.5 inches in barrel length, so we'll err on the side of a +1 here. |
| USP .45 | 7.9 Inches | +1 | +1 |  |
| UMP45 | 8 Inches | +1 | 0 |  |
| Thompson Submachine Gun | 10.5 Inches | +2 | +1 |  |
| Homemade Hand Cannon | Unknown | +2 | +3 | The only thing that can be said for sure is that it's bigger than a Desert Eagle, which makes it pretty dang big for a handgun. But it's also a homemade piece of scrap, so its tolerances are likely poor. |
| L1820 Long Ranger | 20 Inches? | +4 | 0 | If it's based off the Winchester 94, it's likely got a 20 inch barrel. For sanity's sake though, let's cap it at +4. |
| **.44 Magnum** | **Base: 7.5** |  |  |  |
| Desert Eagle .44 | 6 Inches | -1 | 0 | Technically, the Mark XIX Desert Eagle can come with either a 6 Inch (-1) or 10 Inch (+1) barrel, but 6 Inches seems to be the more common barrel |
| LeMat Revolver | 6 Inches? | -1 | 0 | I found figures for the Baby LeMat, but not the full-size one. Doing some measurements on some pictures gave me a figure of 6 Inches, so unless someone can find an actual number here, that's what we'll go with. |
| S&W 629 | 6 Inches | -1 | 0 |  |
| Ruger Redhawk | 7.5 Inches | 0 | 0 | 7.5 inches is being used to match the Super Redhawk (which was supposed to be its successor). |
| **Other Cartridges** | **See Notes** |  |  | **Each gun will have the cartridge's baseline listed in the notes.** |
| Taurus Raging Bull | 8 Inches | 0 | +6 | Baseline: 7.5 Inches |
| S&W 500 | 8.4 Inches | 0 | +8 | Baseline: 8.4 Inches |
## 
## Rifles

The main thing of note here is 5.56 NATO (or as its referred to in-game, .223 Remington). Before the rebalance, it actually made up the entire spread for rifles! The poor maligned M249 clocked in at a pitiful -6 damage, while the slightly smaller M27 IAR packed a whopping +8. That's a 15 point spread! Post-rebalance, it's narrowed down by over half to a 7 point spread. Overall, most rifles tend to trend downward, which makes sense when you think about the emphasis on portability and close-quarters capability in modern times; modern assault rifles and machine guns are smaller and designed to be easier to handle than the older ones. Most of these are still going to be very deadly; they just have some of the damage inflation they were getting hit with removed.

As before, **DmgMod Range (New/Old)** is the range of damage modifiers on the guns, with a comparison of after and before. **Crt. Range (New/Old)** is that same range with the cartridge's actual damage applied, giving an idea of the ''actual'' damage spread when using guns from that caliber.

| Cartridge | Base Length | DmgMod Range (New/Old) | Crt. Range (New/Old) | Notes |
| --- | --- | --- | --- | --- |
| **OVERALL** |  | **-5 to +2 / -6 to +8** |  |  |
| 5.45x39mm | 16.3 Inches | 0 / 0 | 40 / 40 | No changes. |
| 5.56 NATO | 20 Inches | -4 to +2 / -6 to +8 | 42-48 / 40-54 | Includes .223 Remington, but uses 5.56 NATO baseline because all but one of the guns is designed for 5.56 NATO in real life. Without the Leadworks revolver and pipe rifle, it's a -2 to +2 spread. |
| 7.62x39mm | 16.3 Inches | 0 to +1 / 0 to +2 | 36-37 / 36-38 |  |
| 7.62x54mmR | 28 Inches | -3 to 0 / -2 to 0 | 51-54 / 52-54 |  |
| .30-06 | 24 Inches | -3 to 0 / 0 to +7 | 58-61 / 61-68 |  |
| .308 Winchester | 24 Inches | -5 to 0 / -2 to +1 | 63-70 / 68-71 | Drop the pipe rifle, and it becomes a narrower -3 to 0 spread. |
| .300 Winchester Magnum | 24 Inches | 0 / 0 to +5 | 78 / 78-83 |  |
| .700 NX | 28 Inches | 0 / 0 | 90 / 90 |  |
| .50 BMG | 45 Inches | -5 to 0 / +1 | 65-70 / 71 | This is another that appears not to have been adjusted by Stalblar. |

There's not much to say regarding the following table that hasn't already been said. Worth noting, however, is the massive nerfs the Pipe Rifle .30-06 and Browning Automatic Rifle got hit with. Albeit it's only massive because they were rather ridiculously inflated in the first place.

| Weapons | Length (in) | NewDmg | OldDmg | Notes |
| --- | --- | --- | --- | --- |
| **5.45x39mm** | **Base: 16.3** |  |  | No Changes; Skipped. |
| **5.56 NATO** | **Base: 20** |  |  |  |
| L2037 Backup | 9 Inches? | -4 | -2 | Most real-life autorevolvers use 6-7.5 inch barrel, but this one's mentioned to be particularly massive. A 9-inch barrel sounds reasonable in this case. |
| Survivor Carbine | 11 Inches? | -4 | -2 | It's the smallest rifle that isn't a bullpup. So it likely has a pretty short barrel. On top of that, it's a homemade scrap piece. |
| FN SCAR-L | 14 Inches | -2 | +1 |  |
| H&K 416A5 | 14.5 Inches | -2 | +4 | This is assumed to be the 14.5-inch model, matching the M4A1. |
| M4A1 | 14.5 Inches | -2 | +4 |  |
| Bushmaster ACR | 16.5 Inches | -1 | +2 |  |
| M27 IAR | 16.5 Inches | -1 | +8 |  |
| L523 Platform | 18 Inches? | -1 | 0 | It's the same volume as a Steyr AUG, but smaller weight, suggesting it's a smaller weapon. |
| L523 Carbine | 18 Inches? | -1 | -2 | See L523 Platform. |
| L523 Battle Rifle | 18 Inches? | -1 | 0 | Same as the other L523s. It doesn't mention a barrel change, anyhow. |
| H&K G36 | 18.9 Inches | 0 | +6 |  |
| AR-15 | 20 Inches | 0 | 0 | This is assumed to be the standard 20-inch model, matching the M16. |
| Steyr AUG | 20 Inches | 0 | -3 |  |
| M249 | 21 Inches | 0 | -6 |  |
| Ruger Mini-14 | 24 Inches | +1 | +4 |  |
| L523-DSR | 26 Inches? | +2 | +2 | Volume, range, and bullpup all point to a very large barrel on this one. |
| L523-LMG | 26 Inches? | +2 | +2 | It's the only Volume 13 5.56 NATO weapon, AND it's a bullpup. However, some volume's probably taken by ammo, gizmos, and heavy-duty furniture. |
| **7.62x39mm** | **Base: 16.3** |  |  |  |
| Beretta ARX-160 | 16 Inches | 0 | +2 |  |
| AKM | 16.3 Inches | 0 | 0 |  |
| SKS | 20 Inches | +1 | 0 |  |
| **7.62x54mmR** | **Base: 28** |  |  |  |
| Mosin-Nagant M44 & M44-EBR | 20.2 Inches | -3 | -2 |  |
| Mosin-Nagant 1891/30 & 1891/30-EBR | 29 Inches | 0 | 0 |  |
| **.30-06** | **Base: 24** |  |  |  |
| Pipe Rifle .30-06 | 24 Inches? | -2 | +7 | It's the same volume as the other rifles, but at the same time, it's a scrap weapon. |
| Browning BLR | 22 Inches | -1 | 0 |  |
| Survivor Special 700 | 24 Inches | 0 | 0 |  |
| Browning Automatic Rifle | 24 Inches | 0 | +7 |  |
| M1 Garand | 24 Inches | 0 | 0 |  |
| M1903 Springfield | 24 Inches | 0 | 0 |  |
| Remington 700 | 24 Inches | 0 | 0 |  |
| **.308 Winchester** | **Base: 24** |  |  |  |
| Pipe Rifle .308 | Unknown | -5 | 0 | It's the smallest of the .308 firearms, and on top of that, it's a scrap weapon. This means it's a short, handmade barrel, resulting in a considerable loss of energy. |
| FN SCAR-H | 16 Inches | -3 | +1 |  |
| H&K G3 | 17.7 Inches | -2 | 0 |  |
| FN FAL | 21 Inches | -1 | 0 |  |
| M134D-H Minigun | 22 Inches | -1 | 0 |  |
| M1A | 22 Inches | -1 | 0 |  |
| M14 EBR-RI | 22 Inches | -1 | 0 |  |
| M60 | 22 Inches | -1 | 0 |  |
| Savage 111F | 24 Inches | 0 | 0 |  |
| M240 | 24.8 Inches | 0 | -2 |  |
| **.300 Winchester Magnum** | **Base: 24** |  |  |  |
| M2010 ESR | 24 Inches | 0 | +5 |  |
| Weatherby Mark V | 24 Inches | 0 | 0 |  |
| Winchester Model 70 | 24 Inches | 0 | +2 |  |
| **.700 NX** | **Base: 28** |  |  |  |
| Elephant Gun | Unknown | 0 | 0 | Considering it's a generic gun and not a specific make or model, it's set at baseline. |
| **.50 BMG** | **Base: 45** |  |  |  |
| M107A1 | 29 Inches | -5 | +1 |  |
| .50 Caliber Rifle | 33.8 Inches? | -4 | +1 | It's basically a sawed-off M2 HMG; at a -6 (30%) volume reduction, and assuming some of that came from the mechanism change, a 25% drop in barrel length seems reasonable. |
| M2 Browning HMG | 45 Inches | 0 | +1 |  |
## 

One final note: Aside from a couple typo fixes, this PR makes two other notable alterations. First, it moves the Beretta ARX-160 over to 762.json, and makes it accept magazines. Second, it also alters the documentation in GAME_BALANCE.md to reflect the changes, and adds the collected barrel length baselines to the cartridge listings over there.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
> It's a handmade scrap weapon, albeit it IS a bit bigger than the other .40 S&W guns present.

Don't nerf the scrap weapons' damage and accuracy, they suck enough already. Where possible, assume they're made with damage and accuracy in mind.

**Comment by Nioca** [comment]
> Don't nerf the scrap weapons' damage and accuracy, they suck enough already. Where possible, assume they're made with damage and accuracy in mind.

The main factor for pipe rifles/scrap guns is their volume, in relation to comparable weapons. For example, the 9mm Pipe Rifle actually got a small buff, since it's effectively a crude pistol carbine. The .22 LR Pipe Rifle (not in this PR) also gets a buff as well. However, I do think some consideration needs to be given to quality here: a hack-sawed pipe with a 2x4 for a stock is not going to be able to compete with a comparable factory machined weapon. Scrap weapons aren't something you willingly use, they're last-ditch desperation weapons when you've got bullets but no usable gun to go with with it, where risking losing a hand to a misfire is preferable to getting eaten by that charging mi-Go.

With that being said, I could look into tweaking their numbers a bit. The six-shooter's actually one case where there probably wouldn't be as big a difference, since it's actually more carefully crafted and requires more tools.

(Also, this PR, and the ones intended to follow it, doesn't touch accuracy: it's strictly about damage.)

File: data/json/items/gun/45.json [hunk 1]
```diff
@@ -84,15 +86,17 @@
     "type": "GUN",
     "name": "L39B-45",
     "description": "After the success of the 9mm model, Leadworks LLC released an L39B chambered in .45.  It soon earned a reputation as one of the most intimidating guns on the market, for firer and target alike.",
-    "ammo": "45"
+    "ammo": "45",
+    "ranged_damage": 1
```
Comment (mugling) [inline data/json/items/gun/45.json]: Please follow the existing field ordering rather than adding the fields at random. `ranged_damage` should always come after `ammo`

File: data/json/items/gun/45.json [hunk 2]
```diff
@@ -84,15 +86,17 @@
...
   },
   {
     "id": "l_sp_45",
     "copy-from": "l_sp_9mm",
     "type": "GUN",
     "name": "L39-45",
     "description": "Capitalizing on the L39B-45's reputation as one of the most intimidating pistols on the market, Leadworks LLC manufactured a semiautomatic model as well.  Locking it semi-auto reduced cost and improved performance.",
-    "ammo": "45"
+    "ammo": "45",
+    "ranged_damage": 1
```
Comment (mugling) [inline data/json/items/gun/45.json]: This item is literally just the L39 in a different caliber· There is no reason for increased `ranged_damage` here.

File: data/json/items/gun/9mm.json [hunk 3]
```diff
@@ -237,7 +239,7 @@
     "skill": "pistol",
     "ammo": "9mm",
     "range": 1,
-    "ranged_damage": 1,
+    "ranged_damage": 0, "//": "L39s are based off the Robocop Auto-9, in turn a dolled-up Beretta 93R.",
```
Comment (mugling) [inline data/json/items/gun/9mm.json]: Avoid specifying default values

**Comment by mugling** [comment]
> I'm intending to split this into four different PRs, so as to make it easier to review the changes as we go. If desired, however, I could do a larger PR to finish the rest of them in one sweep.

The numbers look fairly good overall. I'd start by updating the ammo `damage` values for the base ammo types only - the issue of tracers and AP rounds can come later once this has had some play testing. It's probably easier as one PR for the relatively low number of lines.

**Comment by Coolthulhu** [comment]
> Scrap weapons aren't something you willingly use, they're last-ditch desperation weapons when you've got bullets but no usable gun to go with with it, where risking losing a hand to a misfire is preferable to getting eaten by that charging mi-Go.

The problem is, current scrap weapons are (in vast majority of cases) not last-ditch desperation weapons. Because they're too weak for that. By the time you can craft them, you already have much better options, including knife spears, copper knives and even rocks or bare hands.

File: data/json/items/gun/45.json (hunk 2 above)
Comment (Nioca) [inline data/json/items/gun/45.json]: That's actually the result of a recent change on your part: The L39-45 and L39-45B used to be Volume 3 weapons (and still were at the time I started working on this). As volume 2, yeah, 0 damage fits better.

Comment (mugling) [inline data/json/items/gun/45.json]: Either `ranged_damage` of `1` or `0` is acceptable but as per the `description` the two items should be otherwise identical apart from their `ammo` field so as a result you should only need make changes to the `l_sp_9mm` definition.

**Comment by Nioca** [comment]
> The numbers look fairly good overall. I'd start by updating the ammo damage values for the base ammo types only - the issue of tracers and AP rounds can come later once this has had some play testing.

I'm not touching the ammo just yet (although I do have some ideas); this PR/series of PRs is strictly about the guns themselves, and their damage modifiers.

> It's probably easier as one PR for the relatively low number of lines.

I agree, but it does result in a rather massive table to make comparisons on. That's why I wanted to go into it intending to do 4 PRs, with the possibility of narrowing down if there weren't any issues.

> The problem is, current scrap weapons are (in vast majority of cases) not last-ditch desperation weapons. Because they're too weak for that. By the time you can craft them, you already have much better options, including knife spears, copper knives and even rocks or bare hands.

That's more an issue with the scrap weapons' ''availability'' rather than their actual stats. Right now, scrap weapons have some rather steep skill requirements for crafting. But even despite that, the above statement is still blatantly untrue. I actually just ran an informal test pitting survivors untrained in any combat skills against mi-gos, and the ones utilizing pipe rifles typically fared better than the ones wielding improvised (or even some dedicated) melee weapons (with the exception of the knife spear, which has reach, and it still only outperformed the pistol-caliber pipe rifles). Heck, the one wielding the Six Shooter actually stopped the mi-go in its tracks without a single scratch, faring better than a survivor with a **broadsword**, and that's **with** the nerf in place. Even the one with the 2 Shot Special would've fended off a mi-go, if I hadn't been a moron and attempted to reload with it right next to me. The 9mm Pipe Rifle performed poorly, but considering that it took emptying the mag on a Beretta M9 just to chase one off, that may have more to do with 9mm than with Pipe Rifle. I didn't test the 22 LR pipe rifle, but I imagine it'd prove rather ineffective simply due to the round it's using.

I guess my point here is that the horrible ineffectiveness of Pipe Rifles are getting blown a bit out of proportion here, and that the problem lies with their availability, not their performance. If you want Pipe Rifles to be useful, your better avenue is to make them available earlier and more easily, not to buff their actual stats.

TLDR: If it comes down to using a Pipe Rifle or an improvised melee weapon, I'll take the Pipe Rifle every time. Unless it's in .22 LR.

**Comment by Coolthulhu** [comment]
>  I actually just ran an informal test pitting survivors untrained in any combat skills against mi-gos

That's a very artificial condition. 3 in all combat skills is reasonable for a low level simulation, but the weapons of choice against mi-gos at that level are kiting, fire and running, not blind charge.

> with the exception of the knife spear, which has reach

Something tells me the rest of your test was likewise of low quality. Because I just tested a naked broadsword wielder at 0 skill with reasonable minimum of tactics of engaging mi-gos at low level (kiting into shrubs) and didn't get a single scratch.

If you're letting mi-gos attack you while you're wielding a melee weapon, but not doing the same for the ranged weapon option, you're conducting a very restricted and unnatural test that doesn't really say much about balance.

> I guess my point here is that the horrible ineffectiveness of Pipe Rifles are getting blown a bit out of proportion here

If knife rifle outperforms them, there is no blowing out of proportions. At 0 skill levels the stick doesn't outperform the rifle (you'll most likely get tired and die), but 0 skill level is when you engage zombies and not mi-gos. Above that, it's better to just wield a stick or a plank - it will be safer, cheaper and more silent.

**Comment by John-Candlebury** [comment]
Wouldn't it make more sense for 6in to be the baseline for .44? 
Since its the most used barrel length in game.

Not that I think a single point of damage would make much difference.

**Comment by Nioca** [comment]
> That's a very artificial condition. 3 in all combat skills is reasonable for a low level simulation, but the weapons of choice against mi-gos at that level are kiting, fire and running, not blind charge.

> If you're letting mi-gos attack you while you're wielding a melee weapon, but not doing the same for the ranged weapon option, you're conducting a very restricted and unnatural test that doesn't really say much about balance.

The idea here was to compare how weapons performed under open conditions, not how well I could kite. With sufficient shrubbery and careful movement, I could probably cheese a mi-go to death with nothing but a lighter and take no damage. Does that make lighters overpowered in comparison to every other weapon in the game?

I won't claim it was a serious and thorough test, far, **far** from it. I personally just wanted to see if the scrap weapons really were as terrible as people keep claiming they were. Frankly, I don't think they are. They most certainly are not worse* than using a rock or your bare fists, to cite two examples you stated.

*With the exception of the .22 LR Pipe Rifle. But again, that has more to do with .22 LR than with the gun itself.

> If knife spear* outperforms them, there is no blowing out of proportions.

I'm presuming you meant spear here, not rifle. Anyhow, considering how powerful reach attacks are, it's not a surprise it's outperforming low-to-mid-caliber pipe rifles. Heck, it probably outperforms most low-to-mid caliber pistols in general. It allows you to kite on open terrain, and land a large number of hits without any fear of retribution, all without using ammunition (other than a modicum of stamina) or making any real noise.

> Wouldn't it make more sense for 6in to be the baseline for .44? Since its the most used barrel length in game.

Sort of. The muzzle energy numbers we're using are directly tied to a specific barrel length; in .44 Magnum's case, the figure used is for 1570 joules from a 7.5 inch barrel. You certainly could get a different muzzle energy figured tied to a 6 inch barrel, and it'd result in a 0/0/0/+1 array instead of a -1/-1/-1/0, but the cartridge itself would drop a point in damage. So ultimately the same result with slightly prettier output, but with more work involved.

**Comment by mugling** [comment]
> I agree, but it does result in a rather massive table to make comparisons on.

It's less work to review everything together otherwise you're likely to require a fifth PR addressing overall balance. You might want to update your title also from part 1/4 presuming your intention is one PR.

Thanks for writing good quality descriptive commits btw.

**Comment by Nioca** [comment]
> Thanks for writing good quality descriptive commits btw.

No problem. :)

The rest of the rebalance has been moved into this PR. In addition to the rebalance, I also moved the Beretta ARX-160 over to 762.JSON and had it accept AKM magazines (which it does in real life IIRC).

File: data/json/items/gun/762.json [hunk 4]
```diff
@@ -30,6 +30,38 @@
     "flags": ["MODE_BURST", "NEVER_JAMS"]
   },
   {
+    "id": "arx160",
+    "type": "GUN",
+    "symbol": "(",
```
Comment (mugling) [inline data/json/items/gun/762.json]: Field ordering is inconsistent with everything else under `gun`. This is required to make for easier item comparison and also to reduce potential for merge conflicts when multiple authors work on the same files.

Comment (Nioca) [inline data/json/items/gun/762.json]: Yeah, the magazine thing (and possessing a Cutting field) was a tip-off that it had been missed on previous passes, but I didn't realize some of the fields were out of order (it was effectively a straight copy-paste from ranged.json, aside from damage and adding magazines). Lemme fix that real quick... (or as quick as crappy dialup allows, anyhow).

Comment (mugling) [inline data/json/items/gun/762.json]: You need to add `arx160` to `itemgroups/guns.json`, most probably under `guns_obscure` or that's likely to get missed. I've been doing this for each item as it's migrated to the `guns/` subdir

Comment (kevingranade) [inline data/json/items/gun/762.json]: Is there an automated way to reorder the fields, possibly with a jq
expression?

Comment (mugling) [inline data/json/items/gun/762.json]: Yes, I have been using such a `jq` script and then running the results through a formatter and then manually editing to improve the readability. The `jq` script also drops defaulted values. Perhaps I should publish this?
